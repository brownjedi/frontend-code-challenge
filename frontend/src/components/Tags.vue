<template>
  <div :class="styles.tags">
    <div
      :class="styles.tag"
      v-for="tag in tags"
      :key="tag"
      :style="{ color: `var(--${tag.toLowerCase()}-text-color)` }"
    >
      {{ tag }}
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent, useCssModule } from 'vue'

export default defineComponent({
  props: {
    tags: {
      required: true,
      type: Array,
    },
  },
  setup() {
    const styles = useCssModule()

    return { styles }
  },
})
</script>

<style lang="scss" module>
.tags {
  display: flex;
}

.tag {
  text-transform: uppercase;
}

.tags > * + * {
  margin-left: 1rem;
}
</style>
