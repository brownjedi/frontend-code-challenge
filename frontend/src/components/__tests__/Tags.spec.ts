import { shallowMount } from '@vue/test-utils'
import Tags from '@/components/Tags.vue'

describe('components/Tags.vue', () => {
  it('has the expected html structure', () => {
    const wrapper = shallowMount(Tags, {
      props: {
        tags: ['Fire', 'Ice'],
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('ensure that the correct number of tags are rendering correctly', async () => {
    // inputs
    const tags = ['Fire', 'Ice', 'Grass']

    const wrapper = shallowMount(Tags, { props: { tags } })
    expect(wrapper.findAll<HTMLDivElement>('.tag').length).toBe(tags.length)
  })
})
