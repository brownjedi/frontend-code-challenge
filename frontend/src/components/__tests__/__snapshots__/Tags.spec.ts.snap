// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Tags.vue has the expected html structure 1`] = `
<div
  class="tags"
>
  
  <div
    class="tag"
  >
    Fire
  </div>
  <div
    class="tag"
  >
    Ice
  </div>
  
</div>
`;
